### ISSUE TITLE
Simple Boot Client doesn't handle hal links properly

### ISSUE THREAD

**Issue by theonlyguills** [spring-projects/spring-hateoas#302]
I have a spring-data-rest test service that returns valid hal+json with a _links section and a self link, e.g.:

```
{
  "asciiArt" : "...",
  "_links" : {
    "self" : {
      "href" : "http://localhost:8002/ponies/luna"
    }
  }
}
```

Yes. It returns ponies.

If I try to consume this with the client using :

```
		ResponseEntity<Resource<Pony>> ponyResponseEntity = new RestTemplate().exchange("http://localhost:8002/ponies/luna", HttpMethod.GET, null, new ParameterizedTypeReference<Resource<Pony>>() {});

```

I get back the properties of my Pony POJO but the links in the Resource are empty.

I was expecting the Resource to contain the self link in this case. 

I made a very simple service and client to explain the problem. They are at:

[pony-client](https://github.com/theonlyguills/pony-client)
and
[pony-service](https://github.com/theonlyguills/pony-webservice)

**in the bug-report branch.**

You can mvn spring-boot:run them both.

`http://localhost:8002/ponies/luna` gives you the hal+json output of the service.
`http://localhost:8080/luna` gives you the client view of the pony and the self link (which is not there)

**Comment by olivergierke** [comment]
Sounds a lot like you run into [this question](http://stackoverflow.com/questions/23239052/why-does-resttemplate-not-bind-response-representation-to-pagedresources/23271778#23271778)? Does the answer solve your problem?

**Comment by theonlyguills** [comment]
I guess I assumed that the hateoas autoconfiguration (that enables hypermediasupport etc) would register the Jackson2HalModule and the MessageConverter for me.

'Option 2' of the answer on SO does work, but it would be nice if this was the default behavior out of the box, unless there are good reasons not to do it.

Maybe a parameter to @EnableHypermediaSupport to turn this on or off?

**Comment by olivergierke** [comment]
You're creating a `RestTemplate` yourself. There's nothing we can do about those instances. If you let the Spring container manage the `RestTemplate` instance, the fix for #293 will register the necessary infrastructure for declared beans. Care to give `0.17.0.BUILD-SNAPSHOT` a try?

**Comment by theonlyguills** [comment]
I had tried with a spring managed RestTemplate before but not using 0.17.0 or a snapshot. You're right the fix for #293 does what I was asking for.
